src/pdf/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,247 @@
+
+    return {
+        householdSize: `${householdSize} (${convertPlural(
+            householdSize - children.length,
+            "Adult"
+        )} ${children.length} Child${children.length ? "ren" : ""})`,
+        genderBreakdown: `${convertPlural(numberFemales, "Female")} ${convertPlural(
+            numberMales,
+            "Male"
+        )} ${convertPlural(householdSize - numberFemales - numberMales, "Other")}`,
+        numberOfBabies: numberBabies.toString(),
+        ageAndGenderOfChildren: displayList(children.map(getChild)),
+    };
+};
+
+const prepareRequirementSummary = (clientData: Schema["clients"]): RequirementSummary => {
+    let babyProduct: string;
+    if (clientData.baby_food) {
+        babyProduct = "Yes";
+    } else {
+        babyProduct = clientData.baby_food === null ? "Don't Know" : "No";
+    }
+
+    return {
+        feminineProductsRequired: displayList(clientData.feminine_products),
+        babyProductsRequired: babyProduct,
+        petFoodRequired: displayList(clientData.pet_food),
+        dietaryRequirements: displayList(clientData.dietary_requirements),
+        otherItems: displayList(clientData.other_items),
+    };
+};
+
+const getQuantityAndNotes = (
+    row: Schema["lists"],
+    size: number
+): Pick<Item, "quantity" | "notes"> => {
+    if (size >= 10) {
+        return { quantity: row["10_quantity"] ?? "", notes: row["10_notes"] ?? "" };
+    }
+    return {
+        quantity: row[`${size}_quantity` as keyof Schema["lists"]] ?? "",
+        notes: row[`${size}_notes` as keyof Schema["lists"]] ?? "",
+    };
+};
+
+const prepareItemsList = async (householdSize: number): Promise<Item[]> => {
+    const listData = await fetchLists();
+    return listData.map((row): Item => {
+        return {
+            description: row.item_name,
+            ...getQuantityAndNotes(row, householdSize),
+        };
+    });
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const getEndNotes = (_parcelID: string): string => {
+    return "PLEASE DISTRIBUTE WEIGHT EVENLY ACROSS BOXES. Space is valuable! Please don't leave boxes half empty - pack efficiently! BOXES MUST BE PACKED FLAT SO THAT THEY CAN BE STACKED. Do not leave items sticking out of the top. We do have a selection of 'free from' goods as well as vegan and halal products. If you're uncertain about anything, please speak to a member of the team.";
+    // This is hard coded for now. _parcelID is also not used for now.
+};
+
+const prepareData = async (parcelID: string): Promise<ShoppingListPDFProps> => {
+    const { parcelInfo, clientID } = await prepareParcelInfo(parcelID);
+    const { clientData, familyData } = await getClientAndFamilyData(clientID);
+    const itemsList = await prepareItemsList(familyData.length);
+
+    const clientSummary = prepareClientSummary(clientData);
+    const householdSummary: HouseholdSummary = prepareHouseholdSummary(familyData);
```

Comment:
is there a reason for `: Household Summary`? does TS not know this type? what about line above/below?

Reply:
removed